# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: afd6a3f623e1
# Full Hash: afd6a3f623e1b0d9de616c1dfa6f2806e4a587af
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-07-31 04:11:13
# Regressor Bug: 1652144
# File Overlap Count: 1
# Description:
#   Bug 1652144 - Part 2: Consider current BrowsingContextGroup for process selection, r=farre
#   
#   This requires keeping track of the current process used to host documents with a
#   particular remote type loaded in each BrowsingContextGroup. Due to lifecycle
#   oddities, this set is kept separate from the existing subscribers set on
# ==============================================================================

diff -r 3d58dcc5e59f -r afd6a3f623e1 docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Thu Jul 30 20:27:34 2020 +0000
+++ b/docshell/base/BrowsingContext.cpp	Thu Jul 30 20:27:37 2020 +0000
@@ -379,7 +379,7 @@
   uint64_t originId = 0;
   if (aOriginProcess) {
     originId = aOriginProcess->ChildID();
-    aGroup->EnsureSubscribed(aOriginProcess);
+    aGroup->EnsureHostProcess(aOriginProcess);
   }
 
   MOZ_LOG(GetLog(), LogLevel::Debug,